### ISSUE TITLE
Consider contributing to GitLab

### ISSUE THREAD

**Issue by dosire** [ppaulweber/gilapv#1]
Thanks for making this. Instead of patching a GitLab installation would it make sense to just patch GitLab upstream so everyone has this behaviour? If so, consider submitting a merge request.

**Comment by ppaulweber** [comment]
Hello @dosire, I think it would be to soon to start with a integration/upstream-contribution, because besides rendering, resource fetching and some navigation control etc. the viewer is far from a complete design. Furthermore, I'm not aware of the current development status regarding PDF and SVG viewing in the current development version of GitLab. I've found some (outdated) issues, but none of them seem to point in this direction with a working solution. That's why I've created the plugin to let the GitLab admin to decide if an integration is desired or not for such a functionality.

**Comment by dosire** [comment]
@ppaulweber Thanks for your response. I'm personally think that have something half working in Gitlab is better because it would allow others to iterate on it. But let's see what @randx thinks

**Comment by dzaporozhets** [comment]
We added SVG support in 8.5 but we would like to see support of more formats in future. Feel free to submit PDF support to upstream so all users can enjoy it. We can help you make it look integrated

**Comment by ppaulweber** [comment]
@dosire yes of course you are right, I will start with appropriate upstream patches ASAP.
@randx thanks for the status overview ... O.K. I will, I've already seen your changes in the >=8.5 =_blob.html.haml= file for SVG support. I'll notify you if I'm ready with a clean an nice solution to integrate it directly in the upstream. Regarding the "look integrated", you can get a "sneak peek" of the PDF viewer at my GitLab instance: http://git.ppaulweber.com/ppaulweber/gilapv-test/blob/master/files/file.pdf

**Comment by dzaporozhets** [comment]
@ppaulweber thank you!
